### DISCUSSION TITLE
Can't Publish My Shiny Apps to Shinyapps.io

### DISCUSSION THREAD

**Discussion by RamadhanRC** [teamalgoritma/community#412] [Q&A] [answered]
Dear All,

Saya butuh bantuannya. Saya ingin publish shinyapp dashboard yang saya buat ke shinyapps.io tapi selalu error padahal shinyapp dashboardnya sudah bisa di run app. Saya tidak tau terdapat kesalahan dimana, mohon bantuannya

Berikut link shinyapps.io saya :
https://ramadhanrc.shinyapps.io/Company_Salaries_Analysis/

Ini keterangan yang muncul pada consolnya:
Warning in mean.default(X[[i]], ...) :
  argument is not numeric or logical: returning NA
Warning in mean.default(X[[i]], ...) :
  argument is not numeric or logical: returning NA
Warning in mean.default(X[[i]], ...) :
  argument is not numeric or logical: returning NA
Warning in mean.default(X[[i]], ...) :
  argument is not numeric or logical: returning NA
Warning in mean.default(X[[i]], ...) :
  argument is not numeric or logical: returning NA
Warning in mean.default(X[[i]], ...) :
  argument is not numeric or logical: returning NA
`summarise()` has grouped output by 'company_size'. You can override using the `.groups` argument.
`summarise()` has grouped output by 'experience_level'. You can override using the `.groups` argument.
`summarise()` has grouped output by 'salary_m_idr'. You can override using the `.groups` argument.

Berikut adalah satu-satunya plot yang menggunakan mean :
output$plot_3 <- renderPlotly({
    avgsal <- compsal %>% 
      group_by(experience_level, work_year) %>% 
      summarise(mean_salary = mean(salary_in_idr/1000000000)) %>% 
      ungroup() %>% 
      arrange(-mean_salary) %>% 
      mutate(label = glue("Level: {experience_level}
                      Tahun: {work_year}
                      Average Salary: {scales::comma(mean_salary)}"))
    
    
    
    plot3 <- ggplot(data = avgsal,
                                   mapping = aes(x = work_year,
                                                 y = mean_salary,
                                                 text = label,
                                                 color = reorder(experience_level,mean_salary),
                                                 group = reorder(experience_level,mean_salary))) + 
      geom_line()+
      geom_point()+
      labs(x = "Year", y = "Average Salary [IDR in Trillion]",
           color = "Employes\nExperience\nLevel")+
      theme_minimal()+
      theme(legend.position = "top")+
      theme(text = element_text(size = 10, face="bold"))
    ggplotly(p=plot3, tooltip = "text")})

Mohon bantuannya

**Comment by onlyphantom** [comment, marked as answer]
> Ini keterangan yang muncul pada consolnya:
> ```
> Warning in mean.default(X[[i]], ...) :
>     argument is not numeric or logical: returning NA
> ```

This means one or more of your `mean()` function call is performed on vectors or lists containing one or more elements that isn't a numeric (`4`, `12.9`, `-1`, `0`) or boolean (`TRUE`, `FALSE`).

```
y <- data.frame(salary_in_idr=c(1,2,"c", 4))
mean(y$salary_in_idr)

# returns:
[1] NA
Warning message:
In mean.default(y$salary_in_idr) :
  argument is not numeric or logical: returning NA
```

Because you're putting it in a dplyr pipeline, you now have many `[1] NA` in `mean_salary` which you then `ungroup` and `arrange`, before putting them in `scales::comma()`, which at that point would throw another error:

```
y <- data.frame(salary_in_idr=c(1,2,"c",NA,4))
scales::comma(y$salary_in_idr)

Error in x * scale : non-numeric argument to binary operator
```

There could be other reasons why it's not deployed successfully, but from the output it's what is clear so far. Maybe add as extra step in dplyr to remove all non numeric values from the `salary_in_IDR` column?

**Reply by RamadhanRC** [reply]
but no NA or non numeric in salary_in_idr and mean_salary data, what should I do?

**Reply by kinantytasyao** [reply]
Hi Bapak @RamadhanRC 

Mohon maaf baru merespon pertanyaan Bapak, setelah memproses datanya sepertinya data Bapak pada salary_in_idr memang memiliki missing values seperti berikut:
![image](https://user-images.githubusercontent.com/79038260/200491206-45f9977d-0500-4f20-9542-e8e0c55e55c2.png)

Mungkin Bapak bisa menangani permasalahan missing values tersebut sehingga warning di atas tidak muncul lagi. Demikian yang dapat kami sampaikan, semoga membantu ya Bapak :)

**Comment by RamadhanRC** [comment]
Is the dbl not a numeric data type?

**Reply by kinantytasyao** [reply]
dbl identical dengan numeric Pak Rizky, silakan bisa melihat dokumentasi berikut jika ingin mengeksplorasi terkait hal ini https://stat.ethz.ch/R-manual/R-devel/library/base/html/double.html
